### PR TITLE
Tracer enhancements

### PR DESCRIPTION
**Pull request by jssblck**
# Overview

- No longer cares what format the NDJSON files are in.
- Logs the count of inspected files.
- Optionally logs the path of each inspected file.

Changes made while doing user troubleshooting; context: https://teamfossa.slack.com/archives/C0155DTGWB1/p1695822830806699?thread_ts=1695310279.254369&cid=C0155DTGWB1

### CODE REVIEW

File: tracer/src/cmd/diagnose.rs [hunk 1]
```diff
@@ -100,6 +103,19 @@ pub fn main(args: Arguments, report_trace_level: Level) -> anyhow::Result<()> {
                 device_ids.insert(id.to_owned());
             }
 
+            if message == "is_file: true" {
```
Comment (csasarak) [inline tracer/src/cmd/diagnose.rs:136]: [Question] This feels really brittle to me. Usually when I write free text in something like a message field I don't usually expect it to mean anything to anyone but humans. I'd expect it to look at something more like a `type` field. 

Is it that this field auto-generated by the tracing library? I'd be more comfortable with it that way.

This just being expedient for now is a good reason to disregard this note also. I just thought I'd ask.

**Review by csasarak — APPROVED** [review]

File: tracer/src/cmd/diagnose.rs (hunk 1 above)
Comment (jssblck) [reply tracer/src/cmd/diagnose.rs:136]: I agree completely- ideally yes, we'd have this in a standardized field instead. Unfortunately, `diagnose` doesn't currently output this information in a standardized field :(

Comment (csasarak) [reply tracer/src/cmd/diagnose.rs:136]: Would you maybe be able to add a comment saying that, or even just linking to this Q/A? NP if you merge it without that though.

Comment (jssblck) [reply tracer/src/cmd/diagnose.rs:136]: done, good callout!

- https://github.com/fossas/foundation-libs/pull/19/commits/5e1e73b01c7fca59ca95c28d347a7b003cf19eae
- https://github.com/fossas/foundation-libs/pull/19/commits/c27e75599ff2e2affe18b7a19f4989bdc02bec96

**Pull request merged**